packages/headless/src/utils/expression-builder.ts

```diff
@@ -0,0 +1,30 @@
+function createExpressionBuilder(expression = '') {
+  const parts = expression ? [expression] : [];
+
+  return {
+    addStringFieldExpression(
+      field: string,
+      operator: '==' | '!=',
```

Comment:
It's a possible approach, but doing the same for numeric and date fields would lead to a large number of methods, which I think will be overwhelming.

If the issue is the `!=` symbols, I prefer to just adjust the string type of `operator` to `equal`, `not equal`, `contains` etc.